data/metadata.json

```diff
@@ -0,0 +1,5823 @@
+      "string_is_multiline": false,
+      "string_pattern": "^(([A-Za-z]{2,3}(-[A-Za-z]{3}(-[A-Za-z]{3}){0,2})?|[A-Za-z]{4,8})(-[A-Za-z]{4})?(-([A-Za-z]{2}|[0-9]{3}))?(-([A-Za-z0-9]{5,8}|[0-9][A-Za-z0-9]{3}))*(-[A-WY-Za-wy-z0-9](-[A-Za-z0-9]{2,8})+)*(-[Xx](-[A-Za-z0-9]{1,8})+)?|[Xx](-[A-Za-z0-9]{1,8})+|[Ii]-[Dd][Ee][Ff][Aa][Uu][Ll][Tt]|[Ii]-[Mm][Ii][Nn][Gg][Oo])$",
+      "examples": [
+        "de",
+        "en",
+        "fr",
+        "frc",
+        "jp"
+      ],
+      "mandatory": {
+        "csaf_base": "nice_to_know",
+        "csaf_security_incident_response": "nice_to_know",
+        "csaf_informational_advisory": "nice_to_know",
+        "csaf_security_advisory": "want_to_have",
+        "csaf_vex": "want_to_have"
+      }
+    },
+    {
+      "path": "$.document.notes",
+      "user_documentation": {
+        "specification": "docs/user/document/notes-spec.en.md",
+        "usage": "docs/user/document/notes-usage.en.md"
+      },
+      "label": "DocumentNotesLabel",
+      "description": "DocumentNotesDescription",
+      "type": "ARRAY",
+      "mandatory": {
+        "csaf_base": "want_to_have",
+        "csaf_security_incident_response": "mandatory",
+        "csaf_informational_advisory": "mandatory",
+        "csaf_security_advisory": "best_practice",
+        "csaf_vex": "want_to_have"
+      }
+    },
+    {
+      "path": "$.document.notes[]",
+      "user_documentation": {
+        "specification": "docs/user/types/notes/note-spec.en.md",
+        "usage": "docs/user/types/notes/note-usage.en.md"
+      },
+      "label": "NotesItemsLabel",
+      "description": "NotesItemsDescription",
+      "type": "OBJECT",
+      "mandatory": {
+        "csaf_base": "",
+        "csaf_security_incident_response": "",
+        "csaf_informational_advisory": "",
+        "csaf_security_advisory": "",
+        "csaf_vex": ""
+      }
+    },
+    {
+      "path": "$.document.notes[].audience",
+      "user_documentation": {
+        "specification": "docs/user/types/notes/note/audience-spec.en.md",
+        "usage": "docs/user/types/notes/note/audience-usage.en.md"
+      },
+      "label": "NotesItemsAudienceLabel",
+      "description": "NotesItemsAudienceDescription",
+      "type": "STRING",
+      "string_is_multiline": false,
+      "string_min_length": 1,
+      "examples": [
+        "all",
+        "executives",
+        "operational management and system administrators",
+        "safety engineers"
+      ],
+      "mandatory": {
+        "csaf_base": "optional",
+        "csaf_security_incident_response": "nice_to_know",
+        "csaf_informational_advisory": "nice_to_know",
+        "csaf_security_advisory": "nice_to_know",
+        "csaf_vex": "nice_to_know"
+      }
+    },
+    {
+      "path": "$.document.notes[].category",
+      "user_documentation": {
+        "specification": "docs/user/types/notes/note/category-spec.en.md",
+        "usage": "docs/user/types/notes/note/category-usage.en.md"
```

Comment:
The usage might be composed of different files (e.g. generic note and document specific note). Maybe, we should use an array? Let's discuss that tomorrow...

Reply:
As discussed in our last meeting we now use an object for the `usage` documentation. It has a mandatory field `generic` linking to the general usage documentation of a field. Optionally, a `specific` field can be used to link to a context specific documentation (in case of re-used types) and profile specific fields `csaf_...` linking to profile specific usage documentation.

See this example:

https://github.com/secvisogram/secvisogram/blob/611fc0617bb76e3cda4ef4785c76e831b88bd3b3/data/metadata.json#L2017-L2021

What is *not* supported in this format is usage that is specific to a context in combination with a profile but I don't know if such cases exist.

Reply:
For the record: My statement of not supporting specific usage for context AND profile is not valid. The context is always given by the path and thus any profile specific usage is also already context specific.